### PR TITLE
Make it compatible with docker compose 2.x

### PR DESCRIPTION
**Pull request by aminvakil**
# Description

docker compose 2.x complains with this error: "key cannot contain a space" or 
```
unexpected character "(" in variable name near "whitelist host after N successful deliveries (N=0 to disable whitelisting)\nPOSTGREY_AUTO_WHITELIST_CLIENTS=5\n\n#
```
when `POSTGREY_TEXT` value is something with spaces.

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by georglauterbach** [comment]
Hey @aminvakil, thanks for providing a PR :) We aim to maintain DMS for Docker Compose version 3. Therefore, I'm not sure whether we want to have compatibility fixes with Compose version 2. @casperklein @wernerfred your opinion on this?

**Comment by aminvakil** [comment]
I'm not sure if I was clear enough or I'm misunderstanding :)

I'm talking about https://github.com/docker/compose/releases/tag/v2.0.1 which has just been released.

**Comment by casperklein** [comment]
I had also issues with the latest docker-compose 2 version and some compose files, but no time yet to take a deeper look.

If white spaces are an issue, we should quote them.

**Comment by aminvakil** [comment]
> If white spaces are an issue, we should quote them.

Thanks, this is definitely a better approach, changed.

**Comment by casperklein** [comment]
The quotes must surround the whole string. Otherwise, the quotes are part of the variable value.

File: mailserver.env [hunk 1]
```diff
@@ -341,7 +341,7 @@ POSTGREY_DELAY=300
 # delete entries older than N days since the last time that they have been seen
 POSTGREY_MAX_AGE=35
 # response when a mail is greylisted
-POSTGREY_TEXT=Delayed by Postgrey
+POSTGREY_TEXT="Delayed by Postgrey"
```
Comment (casperklein) [inline mailserver.env:344]:
```suggestion
"POSTGREY_TEXT=Delayed by Postgrey"
```

**Comment by georglauterbach** [comment]
I didn't even know about this - feels like ages since I left Compose :D Nice!

**Review by georglauterbach — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Comment by casperklein** [comment]
Damn. We should also quote the remaining ENVs, to have them streamlined.

**Comment by casperklein** [comment]
Damn. I've mistaken it.

You need to surround the whole env string in `docker-compose.yml`. However, not when using an extra env_file 😞

**Comment by georglauterbach** [comment]
@casperklein just provide a follow up PR:)

**Comment by casperklein** [comment]
That's weird. I will take a deeper look later.

The documentation says, to quote the whole string, even in an env_file. In my test with docker-compose 1.29.2 however, the var is completely ignored, when full quoted.

Two possibilities:

1. This is a breaking change between the docker-compose versions
2. The documentation is wrong.

**Pull request merged**